docs/core-concepts/versioning.md

```diff
@@ -1,6 +1,201 @@
-# 🔧 .NET Core Versioning
+.NET Core Versioning
+==================== 
 
-> **Note**
-> 
-> This topic hasn’t been written yet! You can track the status of this issue through our public GitHub issue tracker. Learn how you can contribute on GitHub.
+The .NET Core platform is built and versioned in layers. Each of of these layers can be versioned separately for product agility and to accurately describe product changes. While there is significant versioning flexibility, there is a desire to version the platform as a unit to make the product easier to understand.
 
+The product is somewhat unique, being described and delivered via a package manager (NuGet) as packages. While you typically acquire .NET Core as a standalone SDK, the SDK is largely a convenience experience over NuGet packages and therefore not distinct from packages. As a result, versioning is first and foremost in terms of packages and other versioning experiences follow from there.
+
+Semantic Versioning
+===================
+
+.NET Core uses [Semantic Versioning (SemVer)](http://semver.org/), adopting the use of major.minor.patch versioning, using the various parts of the version number to describe the degree and kind of change. This applies to both meta packages (e.g. Microsoft.NETCore.Library) and regular packages (e.g. System.Collections).
+
+Versioning Form
+---------------
+
+MAJOR.MINOR.PATCH[-PRERELEASE.BUILDNUMBER]
+
+Decision Tree
+-------------
+
+MAJOR when:
+  - drop support for a platform
+  - adopt a newer MAJOR version of an existing dependency 
+  - turn a quirk off by default
+
+MINOR when:
+  - add public API surface area 
+  - add new behavior
+  - adopt a newer MINOR version of an existing dependency
+  - introduce a new dependency 
+  
+PATCH when:
+  - make bug fixes
+  - add support for a newer platform
+  - adopt a newer PATCH version of an existing dependency
+  - anything other change
+
+When determining what to increment when there are multiple changes, choose the highest kind of change.
+
+Versioning Scheme
+=================
+
+.NET Core can be defined as and will version in the following way:
+
+- A runtime and framework implementation, distributed as packages. Each package is versioned independently.
+- A set of meta-package packages that reference and make the fine-grained packages easier to reference and use together as a versioned unit. Each meta-package has its own versioning scheme.
```

Comment:
I agree @richlander. Maybe you can rewrite this section to read something like this:

Each .NET Core *choose_a_noun* (component? part? deliverable?) has a different versioning scheme. The following describes how they will version:

- Packages: they contain the runtime and framework implementation. Each package is versioned independently.
- Metapackage packages: they reference (?) and make the fine-grained packages easier to reference and use together as a versioned unit. Each metapackage has its own versioning scheme.
...